lib/Varien/Data/Collection/Db.php

```diff
@@ -586,11 +586,14 @@ public function load($printQuery = false, $logQuery = false)
 
         if (is_array($data)) {
             foreach ($data as $row) {
+                /** @var Mage_Core_Model_Abstract $item */
```

Comment:
```suggestion
```
`$item` does not have to be `Mage_Core_Model_Abstract`, object class is determined by `$_itemObjectClass`